apps/election-manager/src/utils/semsTallies.ts

```diff
@@ -265,6 +261,90 @@ export function convertSEMSFileToExternalTally(
       })
     }
   })
+  return parsedRows
+}
+
+export function parseSEMSFileAndValidateForElection(
+  fileContent: string,
+  election: Election
+): string[] {
+  const errors: string[] = []
+
+  const parsedRows = parseFileContentRows(fileContent)
+
+  if (parsedRows.length === 0) {
+    return [
+      'No valid CSV data found in imported file. Please check file contents.',
+    ]
+  }
+
+  for (const row of parsedRows) {
+    if (!election.precincts.map((p) => p.id).includes(row.precinctId)) {
```

Comment:
This kind of thing is unlikely to be slow here because the number of precincts will likely be small, but in general it'd be better not to incur the cost of the `map`:

```suggestion
    if (election.precincts.every(({ id }) => id !== row.precinctId)) {
```